src/api/app/models/workflow/step/branch_package_step.rb

```diff
@@ -95,7 +105,12 @@ def new_pull_request?
           (gitlab_merge_request? && @scm_extractor_payload[:action] == 'open')
       end
 
-      def add_branch_request_file(package:)
+      def updated_pull_request?
+        (github_pull_request? && @scm_extractor_payload[:action] == 'synchronize') ||
+          (gitlab_merge_request? && @scm_extractor_payload[:action] == 'update')
+      end
+
+      def add_or_update_branch_request_file(package:)
         branch_request_file = case @scm_extractor_payload[:scm]
                               when 'github'
                                 branch_request_file_github
```

Comment:
What about a better name like `branch_request_file_options_for_github`?

Reply:
I've used `branch_request_content_for_github` instead, since what the method returns is going to be the content of the file.